.github/workflows/main.yml

```diff
@@ -65,3 +70,15 @@ jobs:
       timeout-minutes: 30
     - name: Run the Python tests
       run: docker exec test_container /PrairieLearn/docker/test_python.sh
+      timeout-minutes: 5
+    - name: Install dependencies on host
+      run: yarn install --frozen-lockfile
+    - name: Build executor image
+      run: ./tools/executor/build.js
+    - name: Login to DockerHub
+      uses: docker/login-action@v1 
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+    - name: Push executor image to Docker registry 
+      run: ./tools/executor/release.js
```

Comment:
Yeah, good question. Our best option is probably going to be to have AMI builds and PL deploys fetch this image from ECR instead of Docker Hub. We could easily automate a local pull from Docker Hub and a push to ECR. If that fails (e.g. if Docker Hub is down or rate limiting us or otherwise borked), we can also have an automated fallback to a local build of the executor image and the subsequent push to ECR.

I'm thinking of this as a future enhancement and not part of and MVP, as so far Docker Hub has been very reliable during deploys.